### PR TITLE
NNBD

### LINKED ISSUE #70: Migrate to null safety

### PR DESCRIPTION
**Pull request by rxlabz**
Hi,
I hope this PR can help to migrate to a null safe release
fixes #70

### CODE REVIEW

**Comment by adamlofts** [comment]
Hi. Thanks for the contribution.

There seem to be lots changes here which are not strictly related to NNBD, or maybe I am not understanding the scope of change required by NNBD. For example

- Changes to travis, github actions
- Lots of reformatting
- Test changes e.g. `var length = 1 + (serverVersion.length as int)`

Are all this changes connected to NNBD?

**Comment by rxlabz** [comment]
Yes, sorry, some changes are totally out of scope.
- I just opened #75 with the first changes, related to the support of unix socket
- I'll remove the GitHub action and the unnecessary reformatting from this branch

**Comment by Schnurber** [comment]
Hi! I would be very happy if this library would work with null safety. I need it for my project and the dependence on `crypto: ^3.0.0` as well. Thank you!

**Comment by rxlabz** [comment]
Hi @adamlofts, here we go, I removed all the out of scope changes !
The change of mock library is due to the removal of `any` and others arguments matchers in the current NNBD mockito ( in the future, mocktail could be merged in mockito  https://github.com/dart-lang/mockito/issues/347 )

File: pubspec.yaml [hunk 1]
```diff
@@ -10,19 +10,19 @@ authors:
...
 dev_dependencies:
-  mockito: ^3.0.0
+  mocktail: ^0.1.1
   options_file:
-    git:
-      url: https://github.com/ferrerj/optionsfile
-      ref: faebe29
-  test: ^1.3.0
-  pedantic: ^1.9.0
+    git: https://github.com/rxlabz/optionsfile
```
Comment (adamlofts) [inline pubspec.yaml:24]: please add a ref:

File: pubspec.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: mysql1
-version: 0.18.1
+version: 0.19.0-null-safety.0
```
Comment (adamlofts) [inline pubspec.yaml:2]: Reading the migration guide I think we can just publish this as `0.19.0`

https://dart.dev/null-safety/migration-guide#package-version

**Comment by adamlofts** [comment]
Thanks!

**Comment by adamlofts** [comment]
I'll publish to pub now.

**Comment by rxlabz** [comment]
thanks Adam !

**Pull request merged**